src/ln/channelmanager.rs

```diff
@@ -447,6 +451,15 @@ macro_rules! break_chan_entry {
 				}
 				break Err(MsgHandleErrInternal::from_finish_shutdown(msg, channel_id, chan.force_shutdown(), $self.get_channel_update(&chan).ok()))
 			},
+			Err(ChannelError::Stale { msg, update}) => {
+				if let Some(update) = update {
+					if let Err(_e) = $self.monitor.add_update_monitor(update.get_funding_txo().unwrap(), update) {
+						unimplemented!();
```

Comment:
Ugh, please don't add more unimplemented!()s in monitor updating. If nothing else this almost certainly results in the chanmon_fail_consistency fuzzer failing.

Reply:
Ok, replace by a panic the break_chan_entry as it should never be taken. If permanent failure, IMO we can close the channel as we have no more any monitors to watch the chain for us..